Test: Welcome, InfoConceptText, ToastContainer 컴포넌트 테스트 코드 추가

Pangho04 opened the pull request:
## 개요

#57

1. `<Welcome>`, `<InfoConceptText>`, `<ToastContainer>` 컴포넌트 테스트 코드를 추가하였습니다.
2. `<ToastContainer>` 컴포넌트 내의 `handleDeleteAllMessageClick()`의 사용되지 않는 `clearTimeout()`을 삭제했습니다.

## 코드 변경 사항

### 1. 테스트 코드 추가 사항
  - `<Welcome>` :
    - `<InfoConceptText>`의 순환여부. 
  - `<ToastContainer>` : 
    - `MessageList` 표시 여부.
    -`deleteAllMessages()` 작동 여부.

### 2. <ToastContainer /> -> `handleDeleteAllMessageClick()` 내의 `clearTimeout()` 삭제.
https://github.com/team-sticky-252/readim-client/blob/4899316971845fa01656959d6512e98210cea3b6/src/components/Toast/ToastContainer.jsx#L10-L21



## 기타 전달 사항

<!---- 리뷰어들에게 기능관련 외에 전달하고 싶은 사항을 작성해주세요. -->

## PR Checklist

<!---- PR이 다음 요구 사항을 충족하는지 확인하세요. -->

- [x] 주어진 Task의 요구사항을 모두 작성했습니다.
- [x] 작성한 PR을 다시 한번 더 검토하였습니다.
- [x] merge 할 브랜치의 위치를 확인하였습니다.

Pangho04 committed:
Style: InfoConceptText 단위 테스트용 속성 추가

Pangho04 committed:
Refactor: 토스트 메시지 모두 삭제 함수 로직 수정

- 실제론 사용되지 않는 clearTimeout()을 삭제했습니다.

Pangho04 committed:
Test: ToastContainer 컴포넌트 테스트 코드 작성

Pangho04 committed:
Test: Welcome, InfoConceptText 컴포넌트 테스트 코드 작성

JieunZoeyOh commented on src/spec/ToastContainer.spec.jsx:
```diff
@@ -0,0 +1,70 @@
+import { describe, it, expect, vi } from "vitest";
+
+import ToastContainer from "../components/Toast/ToastContainer";
+import { act } from "react";
```
```suggestion
import { act } from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";

import ToastContainer from "../components/Toast/ToastContainer";
```
라이브러리와 컴포넌트를 구분하면 좋을 것 같습니다.

Pangho04 replied:
수정했습니다!

JieunZoeyOh commented on src/spec/ToastContainer.spec.jsx:
```diff
@@ -0,0 +1,70 @@
+    expect(container.firstChild).toHaveClass("animate-fade-out-bottom");
+
+    act(() => {
+      vi.advanceTimersByTime(1500);
```
테스트 설명인 0.5초와 맞추면 좋을 것 같아요!

Pangho04 replied:
ToastMessage 컴포넌트의 단일 토스트 메시지 삭제 함수와의 통일성을 위해 모든 토스트 메시지 함수의 setTimeout을 500ms로 수정하였습니다.

JieunZoeyOh reviewed (approved):
고생하셨습니다.

Pangho04 committed:
Style: 모든 토스트 메시지 삭제 함수 및 테스트 코드 수정

- 단일 토스트 메시지 함수와의 통일성을 위해 setTImeout 시간을 1500 -> 500으로 수정하였습니다.
- 위 수정사항에 맞춰 테스트 코드를 수정하였습니다.
- 테스트코드 import문 순서를 변경하였습니다.

dlgnswk commented on src/spec/ToastContainer.spec.jsx:
```diff
@@ -0,0 +1,70 @@
+      />,
+    );
+
+    const deleteAllbutton = screen.getByText("모두 지우기").parentElement;
```
```suggestion
    const deleteAllButton = screen.getByText("모두 지우기").parentElement;
```
카멜 케이스 적용 부탁드립니다!

Pangho04 replied:
적용 완료했습니다

dlgnswk reviewed (changes requested):
코멘트 확인 부탁드립니다!

Pangho04 committed:
Style: 컨벤션 수정

- 카멜케이스 미적용 변수명 수정

dlgnswk reviewed (approved):
고생 많으셨습니다!
